### PR TITLE
#277 loading feedback

### LINKED ISSUE #277: UI Feature - Loading feedback/Notifications on background operations
As a user, I would like to get feedback from the application on the operations performed and user actions.

Description:
This issue is part of the user story to give more UI feedback to the user on the operations he performs in the application. This task specifically targets having loading displays on the UI when the app performs operations in the background, especially when those tasks freeze, stagger or slow down the application.

Acceptance Criteria:
- [x] Mock-ups have been created and approved by the team
- [x] Mock-ups have been updated on Confluence
- [x] Method agreed for the loading feedback and notifications on background operations or confirmation dialogs
- [x] The loading UI feedback is to be used/needed for the following actions:
- [ ] ~~Saving Asset and Asset Type~~
- [ ] ~~Deleting an asset~~
  - [x] Deleting an asset type
  - [x] Generating trained models for a specific asset type on the asset type information page - model tab
  - [x] Add Fade In Fade Out transition when moving from one screen to another.

### PR DESCRIPTION
**Pull request by PaulMicu**
resolves #277

### CODE REVIEW

**Review by TalalBazerbachi — DISMISSED** [review]
I am getting this error when trying to add an asset.
image
this has nothing to do with this pr, dont use the request changes features to report bugs

File: JavaFXUserInterface/src/main/java/utilities/services/DeleteAssetTypeSercice.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.concurrent.Service;
+import javafx.concurrent.Task;
+import javafx.scene.Scene;
+import utilities.UIUtilities;
+
+import static utilities.TextConstants.ASSET_TYPE_LIST_SCENE;
+
+public class DeleteAssetTypeSercice extends Service<String> {
```
Comment (rdsaliba) [inline JavaFXUserInterface/src/main/java/utilities/services/DeleteAssetTypeSercice.java]: There's a typo in the class name. Should be DeleteAssetTypeService

File: JavaFXUserInterface/src/main/resources/Navigation.fxml [hunk 2]
```diff
@@ -1,37 +1,40 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (rdsaliba) [inline JavaFXUserInterface/src/main/resources/Navigation.fxml:2]: This file needs to be synchronized with Jeff's last pull request. Should not contain any inline style.
More specifically the textFill and font styling should be removed and use CSS instead. @jeffbuonamici do you see anything else?

**Review by rdsaliba — COMMENTED** [review]
Hi Paul. Good job for pulling this one off right on time before the deadline. This brings a great enhancement to the app that was much needed. I'm ok with the coding standards, naming, structure, design. Thank you for doing the unit tests too! I left some comments that requires your attention, should not take too long to fix before merging!

File: JavaFXUserInterface/src/main/resources/AssetTypeInfo.fxml [hunk 3]
```diff
@@ -6,16 +6,20 @@
...
             fx:controller="controllers.AssetTypeInfoController">
+
     <children>
         <fx:include source="Navigation.fxml" AnchorPane.bottomAnchor="0.0" AnchorPane.topAnchor="0.0"/>
-        <AnchorPane prefHeight="200.0" prefWidth="200.0" styleClass="container" AnchorPane.bottomAnchor="0.0"
+        <AnchorPane fx:id="root" prefHeight="200.0" prefWidth="200.0" styleClass="container"
+                    AnchorPane.bottomAnchor="0.0"
                     AnchorPane.leftAnchor="59.0" AnchorPane.rightAnchor="0.0" AnchorPane.topAnchor="0.0">
             <children>
+                <ProgressIndicator fx:id="progressIndicator" layoutX="33.0" layoutY="33.0" scaleX="10" scaleY="10"
```
Comment (JayChouteau) [inline JavaFXUserInterface/src/main/resources/AssetTypeInfo.fxml]: Inline styling. Should be done in the CSS.

**Review by JayChouteau — CHANGES_REQUESTED** [review]
Nice job on the issue. Really neat and well done changes. I agree with Roy's comments and added 1 myself

**Review by JayChouteau — APPROVED** [review]

**Review by rdsaliba — APPROVED** [review]

**Pull request merged**